### ISSUE TITLE
No entry, BUILDING.md or GETTING-STARTED.md

### ISSUE THREAD

**Issue by smaudet** [github/codeql#6109]
There is no entry point?

There is all this stuff about QL (great, some DSL language that seems to be specific to this repo), I see some python, and something about QL being turned into datalog, but then nothing actually about how this repo *actually* evaluates its QL code?

So I guess its MIT so I know, might as well be rainbow colored shit with no fitness for a particular purpose, but this is essentially non-free software at present, or rather a JUMBLE OF CODE which maybe runs on a third party proprietary platform?

At the *least* say "this cannot run on anything but windoze laptops running vs garbo after you pay for a license on lgtm.com" so its clear what this is for...

**Comment by tausbn** [comment]
You seem very angry. I'm sorry this project is not to your satisfaction.

For what it's worth, you can run the [CodeQL CLI](https://codeql.github.com/docs/codeql-cli/) on the platform of your choice (subject to the stated [license notice](https://codeql.github.com/docs/codeql-cli/getting-started-with-the-codeql-cli/#getting-started-with-the-codeql-cli)). There is no requirement that you use VSCode and the CodeQL extension. It is, however, by far the easiest way to get started with using CodeQL.

As for how code is evaluated, the general scheme is that the CodeQL code is compiled into a format called "DIL" (short for "Datalog Intermediary Language"), which in turn is compiled into relational algebra and then subsequently evaluated. The specifics of this process are closed source for the time being, but you can inspect both the DIL and the RA by asking the CodeQL CLI (or extension) to output debug information. You can read more about the evaluation in the appropriate part of the [CodeQL language reference](https://codeql.github.com/docs/ql-language-reference/evaluation-of-ql-programs/).

**Comment by smaudet** [comment]
"The specifics of this process are closed source for the time being"

OK, well then could you update the README.md to that effect.

Not so much "angry" as, "not willing to stand for nonsense like things being open source but actually not because they require something proprietary". I don't have a problem with you being proprietary, just say it.

**Comment by tausbn** [comment]
My assessment of your anger level was based on the language you employed. Terms like "rainbow colored shit" and "vs garbo" do not strike me as the utterances of a calm, rational individual.

We will take your suggestions under advisement. 

However, if I may, please let me offer you a word of advice: you may find people more receptive to your ideas if you address them politely. You'll catch more flies with honey than with vinegar.

**Comment by smaudet** [comment]
Without getting too far into psycho-social behavior, if you see someone with tattoos walking around in a park holstering say, an AK-47, you react in a certain way.

"Under advisement" - acting like something is what its not and even providing a bunch of doco (which I saw and read, at least on the evaluation of QL, before I opened this ticket), deciding someone is angry because they use a strongly worded metaphor, all speak to a certain attitude and behavior, which I do not appreciate at all (certain scummy under-the-table practices come to mind, not that I'm accusing you of these, persay).

But thank you, for taking it _under advisement_. I was intrigued by the project, suffice it to say this interaction has not impressed me, even if I can admit I was a bit strong with my metaphors.
